ee/vellum_ee/workflows/display/utils/expressions.py

```diff
@@ -349,9 +349,23 @@ def serialize_value(executable_id: UUID, display_context: "WorkflowDisplayContex
         }
 
     if isinstance(value, TriggerAttributeReference):
-        # Generate trigger ID using the same hash formula as in base_workflow_display.py
+        # Generate trigger ID using the appropriate method for each trigger type
         trigger_class = value.trigger_class
-        trigger_id = uuid4_from_hash(trigger_class.__qualname__)
+
+        # Track trigger attribute usage for the current workflow serialization so that
+        # trigger definitions only include attributes referenced within the workflow.
+        trigger_usage = display_context.trigger_attribute_usage.setdefault(trigger_class, set())
```

Comment:
[q] why do we need this?

actually [blocking] - even if the attribute is not in the workflow, we still want it to be serialized and sent to the UI so that the user _can_ reference it if they choose to